project-base/storefront/components/Layout/Header/Cart/CartInHeader.tsx

```diff
@@ -38,97 +45,145 @@ export const CartInHeader: FC = ({ className }) => {
+                                    <div className="flex flex-row justify-between mb-10 lg:hidden pr-1">
+                                        <ExtendedNextLink href={cartUrl} onClick={() => setIsClicked(false)}>
+                                            <ArrowRightIcon className="rotate-180 text-borderAccent w-4" />
+                                        </ExtendedNextLink>
+                                        <span className="text-base">{t('Cart')}</span>
+                                        <RemoveIcon
+                                            className="w-4 text-borderAccent cursor-pointer"
+                                            onClick={() => setIsClicked(false)}
+                                        />
+                                    </div>
+                                    <ul className="relative w-[315px] max-h-[78dvh] m-0 flex list-none flex-col overflow-y-auto p-0 lg:w-[510px] lg:max-h-[50dvh] overflow-auto">
+                                        {isRemovingFromCart && <LoaderWithOverlay className="w-16" />}
+                                        {cart.items.map((cartItem, listIndex) => (
+                                            <CartInHeaderListItem
+                                                key={cartItem.uuid}
+                                                cartItem={cartItem}
+                                                onRemoveFromCart={() => removeFromCart(cartItem, listIndex)}
+                                            />
+                                        ))}
+                                    </ul>
+                                    <div
+                                        className={twJoin(
+                                            'flex pt-5 gap-4',
+                                            shouldDisplayTransportBar ? 'justify-between' : 'justify-end',
+                                        )}
+                                    >
+                                        <FreeTransportRange />
+                                        <Button
+                                            className="rounded-lg"
+                                            size="small"
+                                            onClick={() => router.push(cartUrl)}
+                                        >
+                                            {t('Go to cart')}
+                                        </Button>
+                                    </div>
+                                </>
+                            ) : (
+                                <>
+                                    <span>{t('Your cart is currently empty.')}</span>
+                                    <EmptyCartIcon className={twJoin('w-20')} />
+                                </>
+                            )}
+                        </>
+                    )}
+                </div>
             </div>
-        </div>
+            <Overlay
```

Comment:
I noticed a slight delay when hovering over the button. The cart popup opens first, and then the grey overlay appears after the animation ends.

Reply:
Yes, I fixed it already - it will make small delay on hover after that hover popup and overlay in one step.